syscall/consolefs.c

```diff
@@ -91,44 +91,75 @@ static int _consolefs_ioctl(oe_fd_t* file_, unsigned long request, uint64_t arg)
 {
     int ret = -1;
     file_t* file = _cast_file(file_);
+    uint64_t argin = 0;
+    void* argout = NULL;
+    uint64_t argsize = 0;
 
     if (!file)
         OE_RAISE_ERRNO(OE_EINVAL);
 
-    /*
-     * MUSL uses the TIOCGWINSZ ioctl request to determine whether the file
-     * descriptor refers to a terminal device (such as stdin, stdout, and
-     * stderr) so that it can use line-bufferred input and output. This check
-     * fails when delegated to the host since this implementation opens the
-     * devices by name (/dev/stdin, /dev/stderr, /dev/stdout). So the following
-     * block works around this problem by implementing TIOCGWINSZ on the
-     * enclave side. Other terminal control ioctls are left unimplemented.
-     */
-    if (request == OE_TIOCGWINSZ)
+    switch (request)
     {
-        struct winsize
+        /*
+         * MUSL uses the TIOCGWINSZ ioctl request to determine whether the file
+         * descriptor refers to a terminal device (such as stdin, stdout, and
+         * stderr) so that it can use line-bufferred input and output. This
+         * check fails when delegated to the host since this implementation
+         * opens the devices by name (/dev/stdin, /dev/stderr, /dev/stdout). So
+         * the following block works around this problem by implementing
+         * TIOCGWINSZ on the enclave side.
+         */
+        case OE_TIOCGWINSZ:
         {
-            unsigned short int ws_row;
-            unsigned short int ws_col;
-            unsigned short int ws_xpixel;
-            unsigned short int ws_ypixel;
-        };
-        struct winsize* p;
-
-        if (!(p = (struct winsize*)arg))
-            OE_RAISE_ERRNO(OE_EINVAL);
+            struct winsize
+            {
+                unsigned short int ws_row;
+                unsigned short int ws_col;
+                unsigned short int ws_xpixel;
+                unsigned short int ws_ypixel;
+            };
+            struct winsize* p;
+
+            if (!(p = (struct winsize*)arg))
+                OE_RAISE_ERRNO(OE_EINVAL);
+
+            p->ws_row = 24;
+            p->ws_col = 80;
+            p->ws_xpixel = 0;
+            p->ws_ypixel = 0;
+
+            ret = 0;
+            goto done;
+        }
+
+        // arg: void
+        case OE_TIOCEXCL:
+        case OE_TIOCNXCL:
+            break;
 
-        p->ws_row = 24;
-        p->ws_col = 80;
-        p->ws_xpixel = 0;
-        p->ws_ypixel = 0;
+        // arg: int
+        case OE_TCSBRK:
+        case OE_TCXONC:
+        case OE_TCFLSH:
+        case OE_TIOCSCTTY:
+            argin = arg;
+            break;
 
-        ret = 0;
-        goto done;
+        // arg: struct termios*
+        case OE_TCGETS:
+        case OE_TCSETS:
+        case OE_TCSETSW:
+        case OE_TCSETSF:
+            argsize = 60; // sizeof(struct termios)
+            argout = (void*)arg;
```

Comment:
Even though the host side ioctl call is equivalent if the argument is passed via `argin` or `argout`, for code readibility here its a bit confusing that set commands like TCSET* are passing the argument via `argout`.

Reply:
What do you think about renaming `argout` to something like `pointer_arg` (and rename `argin` to `value_arg`) everywhere?